### PR TITLE
feat: deleted prefer-await-to-then

### PR DESCRIPTION
**Pull request by bob606**
Deleted typescript-prefer-await as it is too opinionated about where to use await.

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -490,7 +490,6 @@ module.exports = {
     // * it's better to use one syntax so the await syntax should be our standard
     'promise/prefer-await-to-callbacks': 'warn', // https://github.com/xjamundx/eslint-plugin-promise/blob/master/docs/rules/prefer-await-to-callbacks.md
     // * it's better to use one syntax so the await syntax should be our standard
-    'promise/prefer-await-to-then': 'warn', // https://github.com/xjamundx/eslint-plugin-promise/blob/master/docs/rules/prefer-await-to-then.md
```
Comment (timidak) [inline base.js:493]: I would remove the comment, too.

File: base.js [hunk 2]
```diff
@@ -490,7 +490,6 @@ module.exports = {
     // * it's better to use one syntax so the await syntax should be our standard
```
Comment (timidak) [inline base.js:490]:
```suggestion
    // * it's better to use one syntax so the await syntax should be our standard and in some cases the `then` syntax can be used
```

File: base.js (hunk 1 above)
Comment (timidak) [reply base.js:493]: Also I would leave the rule in the file and just turn it `"off"`. If we change our mind later, we do not have to search for the rule in the docs. What do you mean?

**Review by timidak — CHANGES_REQUESTED** [review]
Just a few changes 🙂

**Comment by bob606** [comment]
please merge if you approve the changes @timidak

**Review by timidak — APPROVED** [review]
👍🏼 nice one!

**Pull request merged**